include/deal.II/base/tensor.h

```diff
@@ -882,9 +893,14 @@ class Tensor
 
 private:
   /**
-   * Array of tensors holding the subelements.
+   * Array of tensors holding the elements of the tensor. If this is
+   * a rank-1 tensor, then we simply need an array of scalars.
+   * Otherwise, it is an array of tensors one rank lower.
    */
-  std::array<Tensor<rank_ - 1, dim, Number>, dim> values;
+  std::conditional_t<rank_ == 1,
+                     std::array<Number, dim>,
+                     std::array<Tensor<rank_ - 1, dim, Number>, dim>>
+    values;
```

Comment:
Can't you just say
```suggestion
  std::array<value_type, dim> values;
```

Reply:
I could, but I don't want to because I want to replace the first `std::array<Number, dim>` by `VectorizedArray` in a follow-up step and then it's not longer an array for both cases.